File: python/examples/basic_usage.py
from agentbay.session_params import CreateSessionParams
from agentbay.exceptions import AgentBayError
from agentbay import AgentBay
import os
import sys

# Add the parent directory to the path so we can import the agentbay package
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))


def main():
    # Initialize the AgentBay client
    # You can provide the API key as a parameter or set the AGENTBAY_API_KEY
    # environment variable
    api_key = os.environ.get("AGENTBAY_API_KEY") or "your_api_key_here"
    session = None
    try:
        agent_bay = AgentBay(api_key=api_key)

        # Create a new session
        # image_id: linux_latest, code_latest, windows_latest etc.
        # use linux_latest because it supports "shell" command
        params = CreateSessionParams(
            image_id="linux_latest",
        )
        print("Creating a new session...")
        session_result = agent_bay.create(params)
        print(f"Session created with ID: {session_result.session.session_id}")
        print(f"Request ID: {session_result.request_id}")

        # Get the session object from the result
        session = session_result.session

        # Execute a command
        print("\nExecuting a command...")
        result = session.command.execute_command("ls -la")
        print(f"Command execution success: {result.success}")
        if result.success:
            print(f"Command output: {result.output}")
        else:
            print(f"Command error: {result.error_message}")

        # Read a file
        print("\nReading a file...")
        result = session.file_system.read_file("/etc/hosts")
        print(f"File reading success: {result.success}")
        if result.success:
            print(f"File content: {result.content}")
        else:
            print(f"File reading error: {result.error_message}")

        # Get session link
        print("\nGetting session link...")
        link_result = session.get_link()
        print(f"Link request ID: {link_result.request_id}")
        print(f"Link: {link_result.data}")

        # List all sessions
        print("\nListing all sessions...")
        sessions_result = agent_bay.list()
        print(f"Available sessions count: {len(sessions_result)}")
        for s in sessions_result:
            print(f"Session ID: {s.session_id}")

        # Delete the session
        print("\nDeleting the session...")
        delete_result = agent_bay.delete(session)
        session = None
        print(f"Delete operation request ID: {delete_result.request_id}")
        print(f"Delete operation success: {delete_result.success}")
        print("Session deleted successfully")

    except AgentBayError as e:
        print(f"AgentBay error: {e}")
        if session:
            try:
                agent_bay.delete(session)
                print("Session deleted during AgentBayError handling")
            except Exception as delete_error:
                print(f"Failed to delete session during error handling: {delete_error}")
    except Exception as e:
        print(f"Unexpected error: {e}")
        if session:
            try:
                agent_bay.delete(session)
                print("Session deleted during other error handling")
            except Exception as delete_error:
                print(f"Failed to delete session during error handling: {delete_error}")


if __name__ == "__main__":
    main()

File: typescript/examples/basic-usage.ts
import { AgentBay, ListSessionParams } from '../src';
import { log, logError } from '../src/utils/logger';
import { getTestApiKey } from '../tests/utils/test-helpers';
async function main() {
  try {
    // Use the test API key function from test-helpers
    const apiKey = getTestApiKey();
    log('Using test API key for demonstration purposes.');

    // Initialize the AgentBay client
    const agentBay = new AgentBay({ apiKey });

    // Create a new session with labels
    log('Creating a new session with labels...');
    const createResponse = await agentBay.create({imageId:'linux_latest'});
    const session = createResponse.data;
    log(`Session created with ID: ${session.sessionId}`);
    log(`Create Session RequestId: ${createResponse.requestId}`);

    // Execute a command
    log('\nExecuting a command...');
    const commandResponse = await session.command.executeCommand('ls -la');
    log('Command result:', commandResponse.data);
    log(`Execute Command RequestId: ${commandResponse.requestId}`);

    // Read a file
    log('\nReading a file...');
    const fileResponse = await session.filesystem.readFile('/etc/hosts');
    log(`File content: ${fileResponse.data}`);
    log(`Read File RequestId: ${fileResponse.requestId}`);

    // Get the session link
    log('\nGetting session link...');
    try {
      const linkResponse = await session.getLink();
      log(`Session link: ${linkResponse.data}`);
      log(`Get Link RequestId: ${linkResponse.requestId}`);
    } catch (error) {
      log(`Note: Failed to get session link: ${error}`);
    }

    // Use the UI module to take a screenshot
    log('\nTaking a screenshot...');
    try {
      const screenshotResponse = await session.ui.screenshot();
      log(`Screenshot data length: ${screenshotResponse.data.length} characters`);
      log(`Screenshot RequestId: ${screenshotResponse.requestId}`);
    } catch (error) {
      log(`Note: Failed to take screenshot: ${error}`);
    }

    // List all sessions by labels using new API
    log('\nListing sessions by labels...');
    try {
      const listParams: ListSessionParams = {
        labels: { test: 'basic-usage' },
        maxResults: 5
      };
      const listResponse = await agentBay.listByLabels(listParams);
      log(`Available sessions count: ${listResponse.data.length}`);
      log(`Total count: ${listResponse.totalCount}`);
      log(`Max results: ${listResponse.maxResults}`);
      log('Session IDs:', listResponse.data.map(s => s.sessionId));
      log(`List Sessions RequestId: ${listResponse.requestId}`);

      // Demonstrate pagination if there's a next token
      if (listResponse.nextToken) {
        log('\nFetching next page...');
        const nextPageParams: ListSessionParams = {
          ...listParams,
          nextToken: listResponse.nextToken
        };
        const nextPageResponse = await agentBay.listByLabels(nextPageParams);
        log(`Next page sessions count: ${nextPageResponse.data.length}`);
        log(`Next page RequestId: ${nextPageResponse.requestId}`);
      }
    } catch (error) {
      log(`Note: Failed to list sessions by labels: ${error}`);
    }

    // Delete the session
    log('\nDeleting the session...');
    const deleteResponse = await agentBay.delete(session);
    log('Session deleted successfully');
    log(`Delete Session RequestId: ${deleteResponse.requestId}`);

  } catch (error) {
    logError('Error:', error);
    // In a Node.js environment, you would use process.exit(1) here
    throw error;
  }
}

main();
